server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java

```diff
@@ -28,12 +41,72 @@ public String getName() {
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(GET, "/_internal/_health"));
+        List<Route> routes = new ArrayList<>();
+        String pathPrefix = getPathPrefix();
+        routes.add(new Route(GET, pathPrefix));
+        if (healthService != null) {
+            Map<String, List<String>> componentsToIndicators = healthService.getComponentToIndicatorMapping();
```

Comment:
You might be able to use a request parameter within the routes that are returned here. Might make this code much easier to read? This way you can just pick out the component id and indicator ids if they exist and forward them on to the transport action?

```
new Route(GET, /_internal/_health/{component_id});

....

request.param("component_id")
```

Reply:
I'm thinking something like how `RestBulkAction` does things

Reply:
Thanks I'll take a look. I knew there had to be a simpler way to deal with this!

Reply:
OK that's done. It led to a slight change in behavior -- now if you give it an unknown component/feature, it just returns a nearly empty response like:
```
{
    "cluster_name": "elasticsearch",
    "components": {}
}
```
rather than throwing an error. I'm not really sure which is preferable, and could make an argument in either direction, which is why I didn't add code to throw an error now.